### PR TITLE
Dish view model added

### PR DESCRIPTION
**Pull request by Hiteshpatel10**
I tried to follow the project structure, please review my PR.
Dish viewmodel is added mention in issue #3 and Dish useCase and repository also added mention in #4.

### CODE REVIEW

File: app/src/main/java/com/example/disher/category/viewmodel/DishViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class DishViewModel @Inject constructor(useCase: IDishesUseCase) : ViewModel() {
+
+    private val _listOfDish: MutableState<List<Dish>> = mutableStateOf(emptyList())
+    val listOfDish: State<List<Dish>> = _listOfDish
+    val useCase = useCase
```
Comment (Kardelio) [inline app/src/main/java/com/example/disher/category/viewmodel/DishViewModel.kt]: Here you dont need to have this line, instead in your constructor add the word `val` before the `useCase` (inside the constructor) then remove this line

File: app/src/main/java/com/example/disher/category/CategoryScreen.kt [hunk 2]
```diff
@@ -28,6 +28,7 @@ fun CategoryScreen(
     viewmodel: CategoryViewModel = hiltViewModel(),
     onCategoryClick: (String) -> Unit
 ) {
+//    val navController by rememberNavController()
```
Comment (Kardelio) [inline app/src/main/java/com/example/disher/category/CategoryScreen.kt]: You can remove this comment that was added please :)

**Comment by Kardelio** [comment]
Hey @Hiteshpatel10 thanks for your changes!
They are literally perfect other than just a few tweaks that I have posted in certain files. Please resolve these simple changes and then i will happily merge! :)

**Review by Kardelio — CHANGES_REQUESTED** [review]
Hey @Hiteshpatel10 thanks for your changes!
They are literally perfect other than just a few tweaks that I have posted in certain files. Please resolve these simple changes and then i will happily merge! :)

**Comment by Hiteshpatel10** [comment]
> Hey @Hiteshpatel10 thanks for your changes! They are literally perfect other than just a few tweaks that I have posted in certain files. Please resolve these simple changes and then i will happily merge! :)

**Review by Kardelio — APPROVED** [review]

**Pull request merged**